experiments/AMIP/components/ocean/slab_ocean_init.jl

```diff
@@ -36,30 +36,33 @@ struct OceanSlabParameters{FT <: AbstractFloat}
 end
 name(::SlabOceanSimulation) = "SlabOceanSimulation"
 
-# init simulation
-function slab_ocean_space_init(::Type{FT}, space, p) where {FT}
+function get_T_sfc(params, coord)
+    T_sfc_0 = FT(params.T_init) #- FT(275) # close to the average of T_1 in atmos
+    anomaly = false
+    anomaly_tropics = true
+    anom = FT(0)
+    radlat = FT(coord.lat / 180 * pi)
+    if anomaly == true
+        lat_0 = FT(60 / 180 * pi)
+        lon_0 = FT(-90 / 180 * pi)
+        radlon = FT(coord.long / 180 * pi)
+        stdev = FT(5 / 180 * pi)
+        anom_ampl = FT(0)
+        anom = FT(anom_ampl * exp(-((radlat - lat_0)^2 / 2stdev^2 + (radlon - lon_0)^2 / 2stdev^2)))
+    elseif anomaly_tropics == true
+        anom = FT(29 * exp(-coord.lat^2 / (2 * 26^2)))
+    end
+    T_sfc = FT(T_sfc_0 + anom)
+    return T_sfc
+end
 
+# init simulation
+function slab_ocean_space_init(space, params)
+    FT = ClimaCore.Spaces.undertype(space)
     coords = ClimaCore.Fields.coordinate_field(space)
 
     # initial condition
-    T_sfc = map(coords) do coord
-
-        T_sfc_0 = FT(p.T_init) #- FT(275) # close to the average of T_1 in atmos
-        anomaly = false
-        anomaly_tropics = true
-        anom = FT(0)
-        radlat = coord.lat / FT(180) * pi
-        if anomaly == true
-            lat_0 = FT(60) / FT(180) * pi
-            lon_0 = FT(-90) / FT(180) * pi
-            radlon = coord.long / FT(180) * pi
-            stdev = FT(5) / FT(180) * pi
-            anom = anom_ampl * exp(-((radlat - lat_0)^2 / 2stdev^2 + (radlon - lon_0)^2 / 2stdev^2))
-        elseif anomaly_tropics == true
-            anom = FT(29) * exp(-coord.lat^2 / (2 * 26^2))
-        end
-        T_sfc = T_sfc_0 + anom
-    end
+    T_sfc = get_T_sfc.(Ref(params), coords)
```

Comment:
Instead of passing the chunky `params` struct, you can pass the only value that you need in `get_T_sfc` (T_sfc_0). Even better, you can start by setting `T_sfc .= T_sfc_0` and then `.+=` a function that computes the anomaly.

Instead of `get_T_sfc`, you'd have `anomaly(coords)`